src/it/scala/ApiServiceISpec.scala

```diff
@@ -0,0 +1,146 @@
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.effect.testing.scalatest.AsyncIOSpec
+import jms4s.JmsAutoAcknowledgerConsumer.AutoAckAction
+import jms4s.config.QueueName
+import jms4s.jms.JmsMessage
+import jms4s.sqs.simpleQueueService
+import jms4s.sqs.simpleQueueService.{ Config, Credentials, DirectAddress, HTTP }
+import org.scalatest.{ BeforeAndAfterAll, BeforeAndAfterEach }
+import org.scalatest.concurrent.{ Eventually, IntegrationPatience }
+import org.scalatest.freespec.AsyncFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import org.typelevel.log4cats.{ LoggerFactory, SelfAwareStructuredLogger }
+import uk.gov.nationalarchives.omega.api.conf.ServiceConfig
+import uk.gov.nationalarchives.omega.api.services.ApiService
+
+import java.nio.file.{ Files, Paths }
+import javax.jms.{ Connection, MessageProducer, Session }
+import scala.concurrent.duration.DurationInt
+import scala.io.Source
+import scala.util.{ Failure, Success, Using }
+
+class ApiServiceISpec
+    extends AsyncFreeSpec with AsyncIOSpec with Matchers with Eventually with IntegrationPatience
+    with BeforeAndAfterEach with BeforeAndAfterAll {
+
+  implicit val loggerFactory: LoggerFactory[IO] = Slf4jFactory[IO]
+  implicit val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+
+  private val apiService = new ApiService(
+    ServiceConfig("temp", 1, 1, 1, 1, "request-general", "omega-editorial-web-application-instance-1")
+  )
+
+  private val replyMessage: String = "reply.json"
+
+  private val jmsClient = simpleQueueService.makeJmsClient[IO](
+    Config(
+      endpoint = simpleQueueService.Endpoint(Some(DirectAddress(HTTP, "localhost", Some(9324))), "elasticmq"),
+      credentials = Some(Credentials("x", "x")),
+      clientId = simpleQueueService.ClientId("ctd-omega-services"),
+      None
+    )
+  )
+  override def beforeAll(): Unit = {
+    val consumerRes = for {
+      client <- jmsClient
+      consumer <-
+        client.createAutoAcknowledgerConsumer(QueueName("omega-editorial-web-application-instance-1"), 1, 100.millis)
+      _ <- Resource.eval(consumer.handle { (jmsMessage, _) =>
+             for {
+               _ <- readTextMessage(jmsMessage)
+             } yield AutoAckAction.noOp
+           })
+    } yield consumer
+    consumerRes.useForever.unsafeToFuture()
+    apiService.start.unsafeToFuture()
+    ()
+  }
+
+  override def afterAll(): Unit =
+    apiService.stop().unsafeToFuture()
+
+  override def afterEach(): Unit =
+    try
+      super.afterEach()
+    finally
+      if (Files.exists(Paths.get(replyMessage))) {
+        Files.delete(Paths.get(replyMessage))
+      }
+
+  private def readTextMessage(jmsMessage: JmsMessage): IO[Unit] =
```

Comment:
More like writing a message, no? 

Then again - why not just update a class level var, and assert that? 

For example:

```
  private var replyMessage: Option[String] = None
```



```
  eventually {
    IO.pure(replyMessage).asserting(_ mustBe Some("The Echo Service says: Hello World!"))
  }
```

```
override def afterEach(): Unit =
  try {
    super.afterEach()
    replyMessage = None
  } finally
    if (Files.exists(Paths.get(replyMessageFilename))) {
      Files.delete(Paths.get(replyMessageFilename))
    }
```

Reply:
No longer uses a `File`, just a `var`, as suggested.

Reply:
Confirmed.